src/index.js

```diff
@@ -183,36 +196,18 @@ const loadDataOnPage = () => {
   loadData((data) => {
     jsonData = data;
 
-    ddb.prefectures = jsonData.prefectures;
-    let newTotals = calculateTotals(jsonData.daily);
-    ddb.totals = newTotals[0];
-    ddb.totalsDiff = newTotals[1];
-    ddb.trend = jsonData.daily;
-    ddb.lastUpdated = jsonData.updated;
-
-    drawKpis(ddb.totals, ddb.totalsDiff);
-    if (!document.body.classList.contains("embed-mode")) {
-      drawLastUpdated(ddb.lastUpdated, LANG);
-      drawPageTitleCount(ddb.totals.confirmed);
-      prefectureTrendCharts = drawPrefectureTable(
-        ddb.prefectures,
-        ddb.totals,
-        prefectureTrendCharts
-      );
-      drawTravelRestrictions(ddb);
-      trendChart = drawTrendChart(ddb.trend, trendChart);
-      dailyIncreaseChart = drawDailyIncreaseChart(
-        ddb.trend,
-        dailyIncreaseChart
-      );
-      prefectureTrajectoryChart = drawPrefectureTrajectoryChart(
-        ddb.prefectures,
-        prefectureTrajectoryChart,
-        LANG
-      );
+    if (!ddb.isLoaded() || jsonData.updated > ddb.lastUpdated) {
+      ddb.previouslyUpdated = ddb.lastUpdated;
+      ddb.lastUpdated = jsonData.updated;
+      ddb.prefectures = jsonData.prefectures;
+      let newTotals = calculateTotals(jsonData.daily);
+      ddb.totals = newTotals[0];
+      ddb.totalsDiff = newTotals[1];
+      ddb.trend = jsonData.daily;
+
+      let event = new Event("covid19japan-redraw");
```

Comment:
I'm using an event to split data loading from rendering. This means that the rendering will happen in a separate macro task and will get scheduled later.